Compression.md

```diff
@@ -0,0 +1,145 @@
+
+An upgrade to WebSocket causes swift-nio to alter the above pipeline in these ways:
+ - the `HTTPDecoder` and `HTTPRespobseEncoder` (and other HTTP related handlers) are removed from the pipeline
+ - an inbound handler [WebSocketFrameDecoder](https://apple.github.io/swift-nio/docs/current/NIOWebSocket/Classes/WebSocketFrameDecoder.html) is added. It convert raw bytes, received on the wire, to WebSocket frames. 
+ - an outbound handler [WebSocketFrameEncoder](https://apple.github.io/swift-nio/docs/current/NIOWebSocket/Classes/WebSocketFrameEncoder.html) is added to convert WebSocket frames to raw bytes to be sent on the wire. 
+
+Additionally, `Kitura-WebSocket-NIO` makes the following changes to the pipeline:
+ - adds `WebSocketConnection`, an inbound handler to process received WebSocket messages
+ - if the `permessage-deflate` negotiation goes through, a `PermessageDeflateCompressor` and `PermessageDeflateDecompressor` are added
+ 
+ The pipelines now look like:
+ - Inbound pipeline: 
+ 
+   *(Operating System)*  
+    \|  
+   `NIOSSLServerHandler`  
+    \|   
+   `WebSocketFrameDecoder`  
+    \|  
+   `PermessageDeflateDecompressor`  
+    \|  
+   `WebSocketConnection`  
+    \|  
+   *(Kitura/WebSocket application)*
+
+ - Outbound pipeline:  
+ 
+   *(Kitura/WebSocket application)*  
+    \|  
+   `PermessageDeflaterCompressor`    
+    \|  
+   `WebSocketFrameEncoder`      
+    \|  
+   `NIOSSLServerHandler`  
+    \|  
+   *(Operating System)*    
+ 
+ [PermessageDeflateCompressor](https://github.com/IBM-Swift/Kitura-WebSocket-NIO/blob/master/Sources/KituraWebSocket/PermessageDeflateCompressor.swift) is an outbound handler used to compress outbound WebSocket messages. [PermessageDeflateDecompressor](https://github.com/IBM-Swift/Kitura-WebSocket-NIO/blob/master/Sources/KituraWebSocket/PermessageDeflateDecompressor.swift) is an inbound handler used to decompress inbound WebSocket messages. Every WebSocket connection where a valid `permessage-deflate` compression was negotiated, gets its own (`PermessageDeflateCompressor`, `PermessageDeflateDecompressor`) pair. 
+ 
+ With this setup, all the inbound data first passes through NIO's `WebSocketDecoder` where the WebSocket frames are built. It then moves into the `PermessageDeflateDecompressor` where multiple frames comprising a message are accumulated and decompressed using `zlib`'s inflater. Subsequently, the decompressed messages are moved to the `WebSocketConnection` handler.
```

Comment:
It would be nice to have an API link for `WebSocketDecoder`.
I've also lost track as to what we're referring to here by NIO's, do you mean SwiftNIO's or something else?

Reply:
Thanks. Added API links to `WebSocketDecoder` and `WebSocketEncoder`.